src/xpcc/architecture/platform/board/olimexino_stm32/olimexino_stm32.hpp

```diff
@@ -0,0 +1,154 @@
+
+//
+// Olimexino STM32 (modified Maple board with STM32F103RBT6)
+// https://www.olimex.com/Products/Duino/STM32/OLIMEXINO-STM32/open-source-hardware
+//
+
+#ifndef XPCC_STM32_OLIMEXINO_STM32_HPP
+#define XPCC_STM32_OLIMEXINO_STM32_HPP
+
+#include <xpcc/architecture/platform.hpp>
+#include <xpcc/debug/logger.hpp>
+#define XPCC_BOARD_HAS_LOGGER
+
+using namespace xpcc::stm32;
+
+
+namespace Board
+{
+
+/// STM32F103RB running at 64MHz generated from the internal 8MHz crystal
+// Dummy clock for devices
+struct systemClock {
+	static constexpr uint32_t Frequency = MHz64;
+	static constexpr uint32_t Ahb = Frequency;
+	static constexpr uint32_t Apb1 = Frequency / 2;
+	static constexpr uint32_t Apb2 = Frequency;
+
+	static constexpr uint32_t Adc1   = Apb2;
+	static constexpr uint32_t Adc2   = Apb2;
+	static constexpr uint32_t Adc3   = Apb2;
+
+	static constexpr uint32_t Spi1   = Apb2;
+	static constexpr uint32_t Spi2   = Apb1;
+	static constexpr uint32_t Spi3   = Apb1;
+
+	static constexpr uint32_t Usart1 = Apb2;
+	static constexpr uint32_t Usart2 = Apb1;
+	static constexpr uint32_t Usart3 = Apb1;
+	static constexpr uint32_t Uart4  = Apb1;
+	static constexpr uint32_t Uart5  = Apb1;
+
+	static constexpr uint32_t Can1   = Apb1;
+
+	static constexpr uint32_t I2c1   = Apb1;
+	static constexpr uint32_t I2c2   = Apb1;
+
+	static constexpr uint32_t Apb1Timer = Apb1 * 2;
+	static constexpr uint32_t Apb2Timer = Apb2 * 1;
+	static constexpr uint32_t Timer1  = Apb2Timer;
+	static constexpr uint32_t Timer2  = Apb1Timer;
+	static constexpr uint32_t Timer3  = Apb1Timer;
+	static constexpr uint32_t Timer4  = Apb1Timer;
+	static constexpr uint32_t Timer5  = Apb1Timer;
+	static constexpr uint32_t Timer6  = Apb1Timer;
+	static constexpr uint32_t Timer7  = Apb1Timer;
+	static constexpr uint32_t Timer8  = Apb2Timer;
+
+	static bool inline
+	enable()
+	{
+		ClockControl::enableInternalClock();	// 8MHz
+		// internal clock / 2 * 16 = 64MHz, => 64/1.5 = 42.6 => bad for USB
+		ClockControl::enablePll(ClockControl::PllSource::InternalClock, ClockControl::UsbPrescaler::Div1_5, 16);
+		// set flash latency for 64MHz
+		ClockControl::setFlashLatency(Frequency);
+		// switch system clock to PLL output
+		ClockControl::enableSystemClock(ClockControl::SystemClockSource::Pll);
+		ClockControl::setAhbPrescaler(ClockControl::AhbPrescaler::Div1);
+		// APB1 has max. 36MHz
+		ClockControl::setApb1Prescaler(ClockControl::Apb1Prescaler::Div2);
+		ClockControl::setApb2Prescaler(ClockControl::Apb2Prescaler::Div1);
+		// update frequencies for busy-wait delay functions
+		xpcc::clock::fcpu     = Frequency;
+		xpcc::clock::fcpu_kHz = Frequency / 1000;
+		xpcc::clock::fcpu_MHz = Frequency / 1000000;
+		xpcc::clock::ns_per_loop = ::round(3000 / (Frequency / 1000000));
+
+		return true;
+	}
+};
+
+// Arduino Footprint
+using A0 = GpioC0;
+using A1 = GpioC1;
+using A2 = GpioC2;
+
+// FIXME - some strangeness
+// C3 doesn't exist in 64 Ball BGA packag, but it does in the lqfp
```

Comment:
This may be an issue with the xpcc device files, where I didn't distinguish between packaging. It's kinda unclear what package the device files is. This has been fixed in modm, but I won't backport this, since it requires changing way too much.